core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -540,6 +542,14 @@ public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<
                                 continue;
                             }
 
+                            // Check if member id is the owner of the offset.
+                            if (!offsetState.getValue().memberId.equals(memberId)) {
+                                log.debug("Member {} is not the owner of offset: {} in batch: {} for the share"
+                                        + " partition: {}-{}", memberId, offsetState.getKey(), inFlightBatch, groupId, topicIdPartition);
+                                throwable = new InvalidRequestException("Member is not the owner of offset");
```

Comment:
I know we are talking about exception handling later today. This one should be `InvalidRecordState`.

Reply:
As discussed in the meeting, I'll change it to `InvalidRecordState`